conda_lock/conda_lock.py

```diff
@@ -148,6 +148,20 @@ def print_proc(proc):
         sys.exit(1)
 
 
+def _process_stdout(stdout):
+    cache = set()
+    for logline in stdout:
+        if logline.endswith("\n"):
+            logline = logline[:-1]
+        if "%" in logline:
+            logline = logline.split()[0]
+            if logline not in cache:
+                yield logline
+                cache.add(logline)
+        else:
+            yield logline
```

Comment:
i could clean this snippet up a bit more, but before i thought i'd ask you what you think of the overall approach.

Reply:
guessing this is an attempt to reduce progressbar spam logging.  Probably with --quiet we might not need this?

Reply:
See https://github.com/conda-incubator/conda-lock/pull/91#issuecomment-818722333